byway/src/main/java/com/google/sps/servlets/DestinationsServlet.java

```diff
@@ -72,21 +71,21 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       logger.atInfo().withCause(e).log("Unable to find UserLocations Entity %s", userKey);
       return;
     }
-    String start= request.getParameter("start-location"); 
+    String start = request.getParameter("start-location"); 
     String destination = request.getParameter("destinations");
     entity.setProperty("start", start);
-    if((ArrayList<String>) entity.getProperty("destinations") == null){
-      ArrayList<String> destinations = new ArrayList<String>();
+    ArrayList<String> destinations = (ArrayList<String>) entity.getProperty("destinations");
+    if(destinations == null){
+      destinations = new ArrayList<String>();
       destinations.add(destination);
       entity.setProperty("destinations", destinations);
     }
     else{
-      ArrayList<String> destinations = (ArrayList<String>) entity.getProperty("destinations");
       destinations.add(destination);
       entity.setProperty("destinations", destinations);
     }
     datastore.put(entity);
-    ArrayList<String> destinations = (ArrayList<String>) entity.getProperty("destinations");
+    destinations = (ArrayList<String>) entity.getProperty("destinations");
```

Comment:
This seems redundant. You just overwrote the datastore entity with the values in the destinations array, so why do you need to then get the information from datastore again? Seems you can just use the value of the destinations array.

Reply:
Haha i have no idea why I did that. I fixed it